src/ngraph/frontend/onnx_import/op/onehot.cpp

```diff
@@ -51,15 +52,13 @@ namespace ngraph
                         std::make_shared<ngraph::op::Slice>(values, Coordinate{1}, Coordinate{2});
                     auto axis = node.get_attribute_value<std::int64_t>("axis", -1);
 
-                    if (axis < 0)
-                    {
-                        axis += indices_shape.size() + 1;
-                    }
-
-                    ASSERT_VALID_ARGUMENT(
-                        node, (axis >= 0) && (axis <= static_cast<int64_t>(indices_shape.size())))
-                        << "invalid 'axis' attribute: "
-                        << node.get_attribute_value<std::int64_t>("axis", -1);
+                    // Accepted range for axis is [-r-1, r] where r = rank(indices). Validate
```

Comment:
```suggestion
                    // Accepted range for axis is [-r, r] where r = rank(indices)+1. Validate
```

Reply:
I think the range should stay [-r-1, r].  `rank(indices)+1` makes the range too wide on the right.

Reply:
Yea you are right.
But now I can see that -r-1 is too wide from the left.
axis = -3
rank = 2
valid_axis => -3 + 2 = -1 which is invalid value

Reply:
Yes, you'er right. The range isn't symmetric. The rank sent to `validation_axis` is rank+1 and the only reason why is increased is because of the wider range on the left. I want to keep comments compatible to the ONNX documentation.